pkg/gameservers/health.go

```diff
@@ -190,15 +190,61 @@ func (hc *HealthController) syncGameServer(key string) error {
 		return nil
 	}
 
-	hc.loggerForGameServer(gs).Info("Issue with GameServer pod, marking as GameServerStateUnhealthy")
-	gsCopy := gs.DeepCopy()
-	gsCopy.Status.State = agonesv1.GameServerStateUnhealthy
-
-	if _, err := hc.gameServerGetter.GameServers(gs.ObjectMeta.Namespace).Update(gsCopy); err != nil {
-		return errors.Wrapf(err, "error updating GameServer %s to unhealthy", gs.ObjectMeta.Name)
+	skip, err := hc.skipUnhealthy(gs)
+	if err != nil {
+		return err
 	}
 
-	hc.recorder.Event(gs, corev1.EventTypeWarning, string(gsCopy.Status.State), "Issue with Gameserver pod")
+	if !skip {
+		hc.loggerForGameServer(gs).Info("Issue with GameServer pod, marking as GameServerStateUnhealthy")
+		gsCopy := gs.DeepCopy()
+		gsCopy.Status.State = agonesv1.GameServerStateUnhealthy
+
+		if _, err := hc.gameServerGetter.GameServers(gs.ObjectMeta.Namespace).Update(gsCopy); err != nil {
+			return errors.Wrapf(err, "error updating GameServer %s to unhealthy", gs.ObjectMeta.Name)
+		}
+
+		hc.recorder.Event(gs, corev1.EventTypeWarning, string(gsCopy.Status.State), "Issue with Gameserver pod")
+	}
 
 	return nil
 }
+
+// skipUnhealthy checks the GameServer is Ready or after Ready, and what the container crash state
```

Comment:
This comment is really hard to read. I tried taking a stab at rewriting it but didn't have any particularly great ways to phrase it. If you can think of a way to say it more clearly that'd be awesome.

Reply:
Took a stab. PTAL when I push my changes up. I think it's better though.